index.bs

```diff
@@ -339,13 +342,66 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "S
 :: A user agent implementing, in conjunction with the underlying platform, the [=Web Authentication API=] and algorithms
     given in this specification, and handling communication between [=authenticators=] and [=[RPS]=].
 
+: <dfn>Credential ID</dfn>
+:: A probabilistically-unique [=byte sequence=] identifying a [=public key credential source=] and its [=authentication assertions=].
+
+    Credential IDs are generated by [=authenticators=] in two forms:
+    1. At least 16 bytes that include at least 100 bits of entropy, or
+    1. The [=public key credential source=], without its [=Credential ID=], encrypted so only the owning authenticator can
+        decrypt it. This form allows the [=authenticator=] to be nearly stateless, by having the [=[RP]=] store any necessary
+        state.
+
+    [=[RPS]=] do not need to distinguish these two [=Credential ID=] forms.
+
 : <dfn>Credential Public Key</dfn>
 :: The public key portion of an [=[RP]=]-specific <dfn>credential key pair</dfn>, generated by an [=authenticator=] and
     returned to an [=[RP]=] at [=registration=] time (see also [=public key credential=]). The private key portion of the
     [=credential key pair=] is known as the <dfn>credential private key</dfn>. Note that in the case of [=self
     attestation=], the [=credential key pair=] is also used as the [=attestation key pair=], see [=self attestation=]
     for details.
 
+: <dfn>Public Key Credential Source</dfn>
+:: A [=credential source=] ([[CREDENTIAL-MANAGEMENT-1]]) used by an [=authenticator=] to generate [=authentication assertions=]. A public key credential source has:
+
+    * A [=Credential ID=].
+    * A [=credential private key=].
+    * The [=Relying Party Identifier=] for the [=[RP]=] that created this credential source.
+    * An optional [=user handle=] in which the [=[RP]'s=] describes the person who created this credential source.
+    * Optional other information used by the authenticator to inform its UI. For example, this might include the user's
+        {{displayName}}.
+
+    The [=authenticatorMakeCredential=] operation creates a [=public key credential source=] bound to an [=authenticator=] and
+    returns the [=credential public key=] associated with its [=credential private key=]. The [=[RP]=] can use this [=credential
+    public key=] to verify the [=authentication assertions=] created by this [=public key credential source=].
+
+: <dfn>Public Key Credential</dfn>
+
+:: A WebAuthn [=public key credential=] is either a [=public key credential source=], the
```

Comment:
overall I like this "public key credential" definintion, thanks for bearing with me :)

I do remain partial, tho, to retaining the opening sentence of "Generically, a credential is data one entity presents to another in order to authenticate the former to the latter [RFC4949].", if that's ok with folks -- i think it appropriately sets the overall context for the reader.

Reply:
Not sure I understand this: here public key credential *could be* equal to a public key credential source.  That confuses me.

Reply:
OK, got it: The term "public key credential" is used for both, the public key and the assertion.

Reply:
perhaps:
s/A webauthn/The term/
s/is either/refers to/
..?

Reply:
Done.